### PR TITLE
Merge 177x into develop 20/10/2020

### PR DESCRIPTION
**Pull request by jolelievre**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Merge 177x into develop 20/10/2020
| Type?         | improvement
| Category?     | ME
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | ~
| How to test?  | ~

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/21522)

### Conflicts

- admin-dev/themes/new-theme/js/fos_js_routes.json
- admin-dev/themes/new-theme/js/pages/order/create/cart-editor.js
- admin-dev/themes/new-theme/js/pages/order/create/create-order-page.js
- admin-dev/themes/new-theme/js/pages/order/create/product-manager.js
- admin-dev/themes/new-theme/js/pages/order/create/shipping-renderer.js
- admin-dev/themes/new-theme/js/pages/order/view/order-view-page.js
- admin-dev/themes/new-theme/public/order.bundle.js
- admin-dev/themes/new-theme/public/order_create.bundle.js
- admin-dev/themes/new-theme/public/order_view.bundle.js
- classes/order/OrderDetail.php
- composer.lock
- controllers/front/ProductController.php
- src/Adapter/Order/AbstractOrderHandler.php
- src/Adapter/Order/CommandHandler/AddCartRuleToOrderHandler.php
- src/Adapter/Order/QueryHandler/GetOrderProductsForViewingHandler.php
- src/PrestaShopBundle/Controller/Admin/ProductController.php
- src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php
- src/PrestaShopBundle/Resources/config/routing/admin/sell/orders/orders.yml
- tests/Integration/Behaviour/Features/Context/CommonFeatureContext.php
- tests/Integration/Behaviour/Features/Context/Domain/OrderFeatureContext.php
- tests/Integration/Behaviour/Features/Scenario/Order/partial_refund.feature
- tests/UI/pages/BO/shopParameters/orderSettings/index.js
- themes/classic/templates/catalog/_partials/product-images-modal.tpl
- themes/classic/templates/checkout/_partials/order-confirmation-table.tpl
- themes/core.js
- themes/core.js.map

### CODE REVIEW

File: classes/Cart.php [hunk 1]
```diff
@@ -332,6 +332,32 @@ public function updateAddressId($id_address, $id_address_new)
         Db::getInstance()->execute($sql);
     }
 
+    /**
+     * Update the Delivery Address ID of the Cart.
+     *
+     * @param int $currentAddressId Current Address ID to change
+     * @param int $newAddressId New Address ID
+     */
+    public function updateDeliveryAddressId(int $currentAddressId, int $newAddressId)
```
Comment (Progi1984) [inline classes/Cart.php:341]:
```suggestion
    public function updateDeliveryAddressId(int $currentAddressId, int $newAddressId): void
```

File: classes/tax/TaxManagerFactory.php [hunk 2]
```diff
@@ -84,6 +84,14 @@ public static function execHookTaxManagerFactory(Address $address, $type)
         return $tax_manager;
     }
 
+    /**
+     * Reset static cache (mainly for test environment)
+     */
+    public static function resetStaticCache()
```
Comment (Progi1984) [inline classes/tax/TaxManagerFactory.php:90]:
```suggestion
    public static function resetStaticCache(): void
```

File: controllers/front/ProductController.php [hunk 3]
```diff
@@ -1335,4 +1349,24 @@ protected function isValidCombination($productAttributeId, $productId)
...
+    {
+        return $this->isQuickView;
+    }
+
+    /**
+     * Set quick view mode.
+     *
+     * @param bool $enabled
+     */
+    public function setQuickViewMode(bool $enabled = true)
```
Comment (Progi1984) [inline controllers/front/ProductController.php:1368]:
```suggestion
    public function setQuickViewMode(bool $enabled = true): void
```

File: src/Core/Domain/Configuration/ShopConfigurationInterface.php [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
```
Comment (Progi1984) [inline src/Core/Domain/Configuration/ShopConfigurationInterface.php:27]:
```suggestion
```

**Comment by Progi1984** [comment]
@jolelievre One Behat test is undefined : https://github.com/PrestaShop/PrestaShop/pull/21522/checks?check_run_id=1278197377

**Review by boubkerbribri — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Comment by PierreRambaud** [comment]
Thanks @jolelievre

**Pull request merged**